utils/migrate_compressed_ddl.sql

```diff
@@ -0,0 +1,81 @@
+-- This utility is for helping to make major DDL changes to compressed 
+-- hypertables. Because you must decompress the entire hypertable, one
+-- might not have sufficient space to do that. An alternate method is to
+-- copy the data from one table to a new table with the correct schema
+-- or constraints, etc. This tool helps you do that chunk by chunk,
+-- compressing as you go to greatly reduce the space requirements to
+-- perform the migration.
+--
+-- Prerequisites:
+--   The target table must be created first, and compression must be
+--   configured for it.
+--    
+--   You must have sufficient space for the compressed data, so roughly
+--   double your current hypertable size, which is usually still
+--   considerably smaller than all the uncompressed data.
+--
+-- Variables:
+--   old_table is the original table to copy from
+--   new_table is the table you are copying to
+--   older_than is the limit for the chunks to recompress. It will
+--   recompress all chunks older than the interval specified
+--
+-- Limitations:
+--   This tool does not re-create policies, continuous aggregates, or
+--   other dependent objects. It only copies the data to a new table.
+--
+-- Usage:
+--   CALL migrate_data('copy_from', 'copy_to', '30 days');
+--
+ 
+
+
+CREATE OR REPLACE procedure migrate_data(
+	old_table text, 
+	new_table text, 
+	older_than interval)
+language plpgsql
+as 
+$$
+DECLARE
+c_row record;
+c_curs cursor for 
+select * from chunknames;
```

Comment:
I think you'll have to open this later given that the temp table isn't created until later in the run, or at least it's a bit confusing.

Reply:
I might even just avoid the temp table all together and use an array, which you can also loop through.

Reply:
I thought the same, but it does actually work. Though I'm not against an array. I'll just move to an array. That would mean I wouldn't need the cursor either anyway.